src/kernel-install/40-early-cpio.install

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+# -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
+# ex: ts=8 sw=4 sts=4 et filetype=sh
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# This file is part of systemd.
+#
+# systemd is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# systemd is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with systemd; If not, see <https://www.gnu.org/licenses/>.
+
+set -e
+
+COMMAND="${1:?}"
+ENTRY_DIR_ABS="${3:?}"
+
+case "$COMMAND" in
+    remove)
+        [ "$KERNEL_INSTALL_LAYOUT" = "bls" ] || exit 0
+
+        for ucode in "${ENTRY_DIR_ABS}"/early-cpio*; do
+            [ -f "$ucode" ] || continue
+            [ "$KERNEL_INSTALL_VERBOSE" -gt 0 ] && echo "Removing $ucode"
+            rm -f "$ucode"
+        done
+        ;;
+    add)
+        for ucode in "intel-ucode.img" "amd-ucode.img" "intel-uc.img" "amd-uc.img" "early_ucode.cpio" "microcode.cpio"; do
```

Comment:
I guess I'm on the record as not being a fan of this weird list from GRUB. :)

I see two solutions here:
1. Template the file like `90-loaderentry.install.in` and have distributions configure this to the names (and maybe together with the line below paths?) of the microcode files they ship.
2. Make this configurable in the kernel-install `install.conf` file.

The latter might make this less brittle, because right now the last matching file that exists will win.